server/src/unifyfs_request_manager.c

```diff
@@ -1440,6 +1440,51 @@ static int process_unlink_rpc(reqmgr_thrd_t* reqmgr,
     return ret;
 }
 
+static int process_get_gfids(reqmgr_thrd_t* reqmgr,
+                             client_rpc_req_t* req)
+{
+
+    int  ret = UNIFYFS_SUCCESS;
+    int* gfid_list;
+    int  num_gfids;
+
+    ret = unifyfs_fops_get_gfids(&gfid_list, &num_gfids);
+    if (ret != UNIFYFS_SUCCESS) {
+        LOGERR("unifyfs_fops_get_gfids() failed");
+    }
+
+    /* send rpc response */
+
+    unifyfs_get_gfids_out_t out;
+
+    /* initialize bulk handle for the gfid_list */
+    hg_size_t segment_sizes[1] = { num_gfids * sizeof(int) };
+    void* segment_ptrs[1] = { (void*)gfid_list };
+    hg_return_t hret = margo_bulk_create(unifyfsd_rpc_context->shm_mid,
+                                         1, segment_ptrs, segment_sizes,
+                                         HG_BULK_READ_ONLY, &out.bulk_gfids);
+    /* Note: unifyfsd_rpc_context defined in margo_server.h */
+    if (hret != HG_SUCCESS) {
+        return UNIFYFS_ERROR_MARGO;
```

Comment:
Need to free ``gfid_list`` in this case?

Reply:
Yep, you're right.  Another good catch.